presto-main/src/test/java/io/prestosql/sql/query/TestHaving.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.sql.query;
+
+import org.testng.annotations.Test;
+
+public class TestHaving
+{
+    @Test
+    public void testImplicitGroupBy()
+    {
+        try (QueryAssertions queryAssertions = new QueryAssertions()) {
+            queryAssertions
+                    .assertQuery(
+                            "SELECT 'x' FROM (VALUES 1, 1, 2) t(a) HAVING true",
+                            "VALUES 'x'");
+        }
```

Comment:
This commit (intentionally) changes behavior. Previously:
```
presto> select a from (values 1,2,3) t(a) having a > 1;
 a
---
 2
 3
```

now the query fails with
```
line 1:8: 'a' must be an aggregate expression or appear in GROUP BY clause
```

Would be good to have an explicit test for this.